tests/test-program.py

```diff
@@ -51,7 +53,87 @@ def execute_command(self, cmd, input_message):  # pylint: disable=no-self-use
             print(errorstr.decode())  # pylint: disable=superfluous-parens
         return exitstatus, outstr.decode(), errorstr.decode()
 
-    def transform_messages(self, testcase, context=None):  # pylint: disable=too-many-locals
+    def read_streaming_output(self, stream_process):
+        """
+        Given a subprocess, read stdout from that process until <EOC> line recieved
+        """
+        proc_stdout = stream_process.stdout
+        buffer = ''
+        itercount = 0
+        eoc_count = 0
+        while not stream_process.poll() and itercount < 1000:
+            itercount += 1
+            next_line = proc_stdout.readline().decode('utf-8').rstrip('\n')
+            if next_line != '<EOC>':
+                buffer += next_line
+            else:
+                eoc_count += 1
+                return json.loads(buffer)
+        err_string = ''.join([x.decode('utf-8') for x in stream_process.stderr.readlines()])
+        raise Exception(err_string)
+
+    def write_streaming_input(self, command, proc_input, p_stdin):
+        """
+        Given a stdin pipe for a subprocess, write command/proc input to CMA subprocess
+        """
+        p_stdin.write((command + "\n").encode('utf-8'))
+        p_stdin.write(json.dumps(proc_input).encode('utf-8'))
+        p_stdin.write('\n'.encode('utf-8'))
+        p_stdin.write('<EOC>\n'.encode('utf-8'))
+        p_stdin.flush()
+
+    def transform_messages_streaming(self, testcase, context=None):
```

Comment:
Very nice. So we're just testing that running the full set of CMA commands in the stream on an input message returns the expected output? And that the stream exits properly

Reply:
Yes, indeed.    We're running the same set of inputs against both stream and individual command flows.  They should have the exact same output (although one obviously runs faster)